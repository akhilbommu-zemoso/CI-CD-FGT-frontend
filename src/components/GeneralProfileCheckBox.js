import React from 'react'
import Grid from '@material-ui/core/Grid';
import Checkbox from '@material-ui/core/Checkbox';
import { makeStyles } from '@material-ui/core/styles';
const useStyles = makeStyles((theme) => ({
    top: {
        marginLeft: 20,
    },
    root: {
        color: 'green',
        '&$checked': {
            color: 'green',
        },
    },
    checked: {},
}));
function GeneralProfileCheckBox(props) {
    const classes = useStyles();
    const [checked, setChecked] = React.useState(true);
    return (
        <div>
            <Grid item xs={6}>
                <div className={classes.top} onClick={props.method}>
                    <Checkbox
                        className={classes.root}
                        checked={props.value}
                        onChange={(event) => props.method(event)}
                        style={{ color: "rgb(4,136,255)" }}
                        name={props.name}
                    />
                </div>
            </Grid>
        </div>
    )
}
export default GeneralProfileCheckBox
