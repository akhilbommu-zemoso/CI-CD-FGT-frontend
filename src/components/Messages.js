import React from 'react'
import GeneralProfileCheckBox from './GeneralProfileCheckBox'
import { makeStyles } from '@material-ui/core/styles';
const useStyles = makeStyles((theme) => ({
    align: {
        display: 'flex',
        fontSize: '14px',
    },
    top: {
        marginTop: 12,
        fontSize: '14px',
    },
    label: {
        marginTop: 10,
        marginBottom: 15,
        textAlign: 'left',
        color: 'rgb(158,160,165)',
        fontSize: '14px',
    },
}));
function Messages(props) {
    const classes = useStyles();
    return (
        <div className={classes.align}>
            <GeneralProfileCheckBox name={props.name} value={props.value} method={(event) => props.method(event)} />
            <div className={classes.top}>
                {props.label}
            </div>
        </div>
    );
}
export default Messages
