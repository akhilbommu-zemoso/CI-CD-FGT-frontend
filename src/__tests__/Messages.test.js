import React from 'react'
import { render,cleanup, fireEvent,within, getAllByText, getByText ,screen} from '@testing-library/react'
import Messages from '../components/Messages'
import GeneralProfileCheckBox from '../components/GeneralProfileCheckBox'
import { BrowserRouter as Router } from 'react-router-dom'

afterEach(cleanup)
const handleChange = jest.fn();
it('should take a snapshot', () => {
    const { asFragment, getByTestId } = render(
        <Router>
            <Messages
            name={"Text messages"}
            label={"Messages"}
            value={true}
            method={handleChange} />
        </Router>)
    expect(asFragment(<Router>
        <Messages
            name={"Text messages"}
            label={"Messages"}
            value={true}
            method={handleChange} />
    </Router>)).toMatchSnapshot()
});

it('check for label', () => { 
    const { getByText, getByTestId } = render(<Router>
        <Messages
        name={"Text messages"}
        label={"Messages"}
        value={true}
        method={handleChange} /></Router>)
    expect(screen.getByText("Messages")).toBeInTheDocument
   
})


