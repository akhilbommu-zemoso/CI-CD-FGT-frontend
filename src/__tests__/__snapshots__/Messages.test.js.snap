// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-align-1"
  >
    <div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
      >
        <div
          class="makeStyles-top-4"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-116 MuiCheckbox-root MuiCheckbox-colorSecondary makeStyles-root-5 PrivateSwitchBase-checked-117 Mui-checked MuiIconButton-colorSecondary"
            style="color: rgb(4, 136, 255);"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                checked=""
                class="PrivateSwitchBase-input-119"
                data-indeterminate="false"
                name="Text messages"
                type="checkbox"
                value=""
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
        </div>
      </div>
    </div>
    <div
      class="makeStyles-top-2"
    >
      Messages
    </div>
  </div>
</DocumentFragment>
`;
